engine/gamesys/src/gamesys/scripts/script_tilemap.cpp

```diff
@@ -197,7 +197,8 @@ namespace dmGameSystem
      * @param y [type:number] y-coordinate of the tile
      * @param tile [type:number] index of new tile to set. 0 resets the cell
      * @param [h-flipped] [type:boolean] optional if the tile should be horizontally flipped
-     * @param [v-flipped] [type:boolean] optional i the tile should be vertically flipped
+     * @param [v-flipped] [type:boolean] optional if the tile should be vertically flipped
+     * @param [rotated] [type:boolean] optional if the tile should be 90 degrees rotated
```

Comment:
Honestly, I feel this api is a bit too confusing.
Is it the best option for the developer?
Sure, this api is for a programmer, and not used by the level designer.

I believe it is much easier to understand if the api take a rotation in degrees: 0, 90, 180, and 270, and then internally we decide what to do with it?

@britzl what do you think?

Reply:
Also, I think the documentation needs to clarify in what order the transformarion happens. For instance:

`ROT90 + HFLIP != HFLIP + ROT90`

Reply:
Clarify that it's 90 degrees Counter Clockwise

Reply:
Yes, while it's clever that you can rotate and flip to achieve all four rotations it is a nightmare for developers and I'm sure that the first thing they will do is to create helper functions to deal with rotations.

I suggest that we define four enums/constants and use that instead. tilemap.TILE_ROTATION_90, tilemap.TILE_ROTATION_180 etc

Maybe we should consider separate functions as well?

* tilemap.set_hflip(url, layer, x, y, flip)
* tilemap.set_vflip(url, layer, x, y, flip)
* tilemap.set_rotation(url, layer, x, y, rotation)

Reply:
>I believe it is much easier to understand if the api take a rotation in degrees: 0, 90, 180, and 270, and then internally we decide what to do with it?

yes, I agree. but I would like to avoid functionality duplication because having both **h** and **v** and **rotation** in degrees, creates many different ways to do the same.

I thought about some helper function like

```
local flip = true
local one, two, three = tilemap.get_rotation(TILE_ROTATION_90, flip)
-- and use it like
tilemap.set_tile("/go2#1", "layer1", 1, 1, 29, tilemap.get_rotation(TILE_ROTATION_90, flip) )
```

or maybe new `set_tile` function (`something` because I dunno how to nate it)
```
tilemap.set_tile_ something(/go2#1", "layer1", 1, 1, 29, TILE_ROTATION_90, flip)
```

>Maybe we should consider separate functions as well?

not sure about separate functions, it makes applying of parameters much harder and feels like overcomplication for me

Reply:
> not sure about separate functions, it makes applying of parameters much harder and feels like overcomplication for me

Yes, maybe. I'm just not very fond of functions with a long list of arguments.

Another idea might be to combine flip+rotation into a bitmask:

`tilemap.set_tile(url, layer, x, y, tile, tilemap.HFLIP + tilemap.ROTATE90)`

> creates many different ways to do the same

Yes, but that's not necessarily a bad thing. In this case it's down to what the developer wants to achieve. If the developer wants to rotate a tile just skip hflip and vflip and use rotation. If the developer wants to flip the tile then use hflip and/or vflip.

Reply:
I'm with Björn here in that I don't want many arguments to a function. Imo ~6-7 are usually an upper limit for people to remember.

I think the bit mask idea could be good. Only special concern might be that we're using the same name, and using less arguments than previously. It will require some extra care in the function, in order to decide if it's the new interface that is being called.

Reply:
should I keep internal API with 3 boolean flags or it would be better to use one uint with flags?

Reply:
I updated method, and now it's possible to use summ of flip and rotation:
```
	local H, V, R90, R180, R270 = tilemap.H_FLIP, tilemap.V_FLIP, tilemap.ROTATE_90, tilemap.ROTATE_180, tilemap.ROTATE_270
	
	tilemap.set_tile("/go2#1", "layer1", 1, 2, 29, 0)		            --0
	tilemap.set_tile("/go2#1", "layer1", 2, 2, 29, H)		            --1 001
	tilemap.set_tile("/go2#1", "layer1", 3, 2, 29, V)		            --2 010
	tilemap.set_tile("/go2#1", "layer1", 4, 2, 29, H+V)		    --3 011
	tilemap.set_tile("/go2#1", "layer1", 5, 2, 29, R90)		    --4 100
	tilemap.set_tile("/go2#1", "layer1", 6, 2, 29, R90+H)   	    --5 101
	tilemap.set_tile("/go2#1", "layer1", 7, 2, 29, R90+V)	            --6 110
	tilemap.set_tile("/go2#1", "layer1", 8, 2, 29, R90+H+V)	    --7 111
	tilemap.set_tile("/go2#1", "layer1", 9, 2, 29, R180)		    --3 
	tilemap.set_tile("/go2#1", "layer1", 10, 2, 29, R180+H)         --2
	tilemap.set_tile("/go2#1", "layer1", 11, 2, 29, R180+V)           --1
	tilemap.set_tile("/go2#1", "layer1", 12, 2, 29, R180+H+V)     --0
	tilemap.set_tile("/go2#1", "layer1", 13, 2, 29, R270)		    --7
	tilemap.set_tile("/go2#1", "layer1", 14, 2, 29, R270+H)	    --6
	tilemap.set_tile("/go2#1", "layer1", 15, 2, 29, R270+V)	    --5
	tilemap.set_tile("/go2#1", "layer1", 16, 2, 29, R270+H+V)	    --4
```